e2e/playwright/actions/src/tests/share/share-file.spec.ts

```diff
@@ -0,0 +1,248 @@
+/*!
+ * Copyright © 2005-2023 Hyland Software, Inc. and its affiliates. All rights reserved.
+ *
+ * Alfresco Example Content Application
+ *
+ * This file is part of the Alfresco Example Content Application.
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail. Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * The Alfresco Example Content Application is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The Alfresco Example Content Application is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * from Hyland Software. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { ApiClientFactory, NodesApi, test, timeouts, Utils, SharedLinksApi } from '@alfresco/playwright-shared';
+import { expect } from '@playwright/test';
+
+test.describe('Share a file', () => {
+  const random = Utils.random();
+
+  const username = `user-${random}`;
+  const parent = `parent-${random}`;
+  let parentId: string;
+
+  const file3 = `file3-${random}.txt`;
+  const file4 = `file4-${random}.txt`;
+  const file5 = `file5-${random}.txt`;
+  const file6 = `file6-${random}.txt`;
+  const file7 = `file7-${random}.txt`;
+  const file8 = `file8-${random}.txt`;
+  const file9 = `file9-${random}.txt`;
+
+  const shareLinkPreUrl = `/#/preview/s/`;
+
+  const apiClientFactory = new ApiClientFactory();
+
+  test.beforeAll(async () => {
+    test.setTimeout(timeouts.extendedTest);
+    await apiClientFactory.setUpAcaBackend('admin');
+    await apiClientFactory.createUser({ username });
+    const nodesApi = await NodesApi.initialize(username, username);
+
+    parentId = (await nodesApi.createFolder(parent)).entry.id;
+  });
+
+  test.afterAll(async () => {
+    await apiClientFactory.nodes.deleteNodes([parentId]);
+  });
+
+  test.describe('when logged out', () => {
+    let file6SharedLink: string;
+    let file6Id: string;
+
+    test.beforeAll(async () => {
+      test.setTimeout(timeouts.extendedTest);
```

Comment:
test SetTimeout will not require if beforeAll require very less time.

Reply:
removed timeouts where beforeAll is executing properly